hibp.js

```diff
@@ -118,9 +118,14 @@ const HIBP = {
       if (sha1.toUpperCase() === sha1Prefix + breachedAccount.hashSuffix) {
         foundBreaches = allBreaches.filter(breach => breachedAccount.websites.includes(breach.Name));
         foundBreaches = this.filterBreaches(foundBreaches);
+
+        // NOTE: DO NOT CHANGE THIS SORT LOGIC
+        // We store breach resolutions by recency indices,
+        // so that our DB does not contain any part of any user's list of accounts
```

Comment:
self-note: we should move this sorting function into a utility function and use it everywhere when we mark breaches as resolved or when we show breaches as resolved/not.

Reply:
One thing we should probably consider is what happens if we add back in other breach types (IsSpamList, etc)?

Reply:
Updated to annotate the breaches with `recencyIndex` BEFORE `filter`ing the breaches.

Reply:
Great catch, BTW!